services/question/src/controllers/questionController.ts

```diff
@@ -1,11 +1,243 @@
 import { Request, Response } from 'express';
+import { handleError, handleNotFound, handleBadRequest, handleSuccess } from '../utils/helpers';
 import { Question } from '../models/questionModel';
 
+/**
+ * This endpoint allows the retrieval of all the questions in the database (or can filter by optional parameters).
+ * @param req
+ * @param res
+ */
 export const getQuestions = async (req: Request, res: Response) => {
-    const questions = await Question.find();
-    const questionTitles = questions.map(q => q.title);
-    res.status(200).json({
-        message: 'These are all the question titles:' + questionTitles,
-    });
-    return;
+    try {
+        const { title, description, topics, difficulty } = req.query;
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        const query: any = {};
+
+        if (title) {
+            query.title = { $regex: `^${title as string}$`, $options: 'i' };
```

Comment:
Not sure why regex is being used here for exact (case-insensitive) matches? Given that the server would be the one calling the service, don't think this is necessary (and it would result in slower DB queries).

Reply:
Ohhh okay I get what you mean already, I will make the change.